Products/CMFPlone/controlpanel/browser/usergroups_groupdetails.py

```diff
@@ -2,13 +2,25 @@
 from plone.base import PloneMessageFactory as _
 from plone.protect import CheckAuthenticator
 from Products.CMFCore.utils import getToolByName
-from Products.CMFPlone.controlpanel.browser.usergroups import (
-    UsersGroupsControlPanelView,
-)
+from Products.CMFPlone.controlpanel.browser.usergroups import UsersGroupsControlPanelView
 from Products.statusmessages.interfaces import IStatusMessage
 
-
 class GroupDetailsControlPanel(UsersGroupsControlPanelView):
+    def extract_invalid_emails(self, emails):
+        """Validate a list of email addresses."""
+        if isinstance(emails, str):
+            emails = [e.strip() for e in emails.split('\n') if e.strip()]
+        
+        invalid_emails = []
+
+        plone_utils = getToolByName(self.context, "plone_utils")
+
+        for email in emails:
+            if not plone_utils.validateEmailAddresses(email):
```

Comment:
This is still not valid... see my previous comment

Reply:
> I'm sorry, I was not sure what to do next.

No need to be sorry!

I think you want to use `validateSingleEmailAddress`. Check this out: https://github.com/plone/Products.CMFPlone/blob/4a765ec624f5b93e6388c2057a8caced954ae958/Products/CMFPlone/PloneTool.py#L158
Also the tests might give you more insights on what you can do: https://github.com/plone/Products.CMFPlone/blob/4a765ec624f5b93e6388c2057a8caced954ae958/Products/CMFPlone/tests/testPloneTool.py#L34

If you want to add a test yourself please do it here: https://github.com/plone/Products.CMFPlone/blob/4a765ec624f5b93e6388c2057a8caced954ae958/Products/CMFPlone/controlpanel/tests/test_controlpanel_usergroups.py

Thanks again for being patient!

Reply:
verify again @ale-rt